scripts/genllheaders/genllheaders.py

```diff
@@ -0,0 +1,101 @@
+"""
+Utility to autogenerate generic LL HAL headers.
+
+Usage::
+
+    python3 genllheaders.py [-p /path/to/HAL] [-o /path/to/output_dir]
+
+Copyright (c) 2020 Teslabs Engineering S.L.
+
+SPDX-License-Identifier: Apache-2.0
+"""
+
+import argparse
+from pathlib import Path
+import re
+import shutil
+
+from jinja2 import Environment, FileSystemLoader
+
+
+SCRIPT_DIR = Path(__file__).absolute().parent
+"""Script directory."""
+
+TEMPLATE_FILE = "header-template.j2"
+"""Template file."""
+
+REPO_ROOT = SCRIPT_DIR / ".." / ".."
+"""Repository root (used for input/output default folders)."""
+
+LL_API_IGNORE = [
+    "usb"
+]
+"""List of LL APIs to be ignored."""
+
+
+def main(hal_path, output):
+    """Entry point.
+
+    Args:
+        hal_path: Zephyr CubeMX HAL path.
+        output: Output directory.
+    """
+
+    if output.exists():
+        shutil.rmtree(output)
+    output.mkdir(parents=True)
+
+    # obtain all available LL APIs for each family
+    ll_apis = dict()
+    for entry in sorted(hal_path.iterdir()):
+        if not entry.is_dir() or not entry.name.startswith("stm32"):
+            continue
+
+        family = entry.name
```

Comment:
To remain in line with usual vocabulary, please use `series` instead of `family`.
For instance: we usually refer as `stm32 family` and `f1 series`.

Reply:
adjusted names